test/e2e/service.go

```diff
@@ -246,6 +248,96 @@ var _ = Describe("Services", func() {
 		}()
 	}, 240.0)
 
+	It("should be able to create a functioning external load balancer", func() {
+		serviceName := "external-lb-test"
+		ns := api.NamespaceDefault
+		labels := map[string]string{
+			"key0": "value0",
+			"key1": "value1",
+		}
+		service := &api.Service{
+			ObjectMeta: api.ObjectMeta{
+				Name: serviceName,
+			},
+			Spec: api.ServiceSpec{
+				Port:                       80,
+				Selector:                   labels,
+				TargetPort:                 util.NewIntOrStringFromInt(80),
+				CreateExternalLoadBalancer: true,
+			},
+		}
+
+		By("cleaning up previous service " + serviceName + " from namespace " + ns)
+		c.Services(ns).Delete(serviceName)
+
+		By("creating service " + serviceName + " with external load balancer in namespace " + ns)
+		result, err := c.Services(ns).Create(service)
+		Expect(err).NotTo(HaveOccurred())
+		defer func(ns, serviceName string) { // clean up when we're done
+			By("deleting service " + serviceName + " in namespace " + ns)
+			err := c.Services(ns).Delete(serviceName)
+			Expect(err).NotTo(HaveOccurred())
+		}(ns, serviceName)
+
+		if len(result.Spec.PublicIPs) != 1 {
+			Failf("got unexpected number (%d) of public IPs for externally load balanced service: %v", result.Spec.PublicIPs, result)
+		}
+		ip := result.Spec.PublicIPs[0]
+		port := result.Spec.Port
+
+		pod := &api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1beta1",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name:   "elb-test-" + string(util.NewUUID()),
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "webserver",
+						Image: "kubernetes/test-webserver",
+					},
+				},
+			},
+		}
+
+		By("creating pod to be part of service " + serviceName)
+		podClient := c.Pods(api.NamespaceDefault)
+		defer func() {
+			By("deleting pod " + pod.Name)
+			defer GinkgoRecover()
+			podClient.Delete(pod.Name)
+		}()
+		if _, err := podClient.Create(pod); err != nil {
+			Failf("Failed to create pod %s: %v", pod.Name, err)
+		}
+		expectNoError(waitForPodRunning(c, pod.Name))
+
+		By("hitting the pod through the service's external load balancer")
+		var resp *http.Response
+		for i := 0; i < 10; i++ {
+			resp, err = http.Get(fmt.Sprintf("http://%s:%d", ip, port))
+			if err == nil {
+				break
+			}
+			if i < 9 {
+				By("failed to hit pod through load balancer, sleeping and trying again")
+				time.Sleep(10 * time.Second)
+			}
+		}
+		Expect(err).NotTo(HaveOccurred())
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
It returns an index page with links to a few sub-pages. Rather than adding the extra overhead of grabbing the hostnames and requiring this to run on a particular one, I've added a check that the string test-webserver is in the body output.